### ISSUE TITLE
Unable to log passwort protected OC cache

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#15344]
### Describe your problem!

Today I wanted to log an OC.de virtual cache, where logging requires a password.
The password field however does not show up in c:geo logging view and thus sending the log fails.

### How to reproduce?

- Open OC1312E
- Create a found log -> No password field shown
- Send the log

### Actual result after these steps?

Log sending fails (as no password was submitted)

### Expected result after these steps?

Log should be send if the correct password is provided

### Reproducible

Yes

### c:geo Version

2024.02.15-NB

### System information

```text
Android 14
```

**Comment by eddiemuc** [comment]
Could not reproduce in emulator based on current master state:

![image](https://github.com/cgeo/cgeo/assets/6909759/74918e9f-05cd-4faf-992d-6ba3e39110cb)

Looking into code, there are three conditions underr which the password field is shown:
* Must be a CREATE log (not a log edit) -> but for OC we don't support log edit anyway
* Must be a FOUND_IT log
* Cache must have the "requires Password" field set

![image](https://github.com/cgeo/cgeo/assets/6909759/26c5ae69-65f0-4394-923b-9a40eb5ffcca)

Changing log type to something else (e.g. "write note") makes password field disappear, changing it back makes it reappear. Just as expected.

@Lineflyer a wild guess: is the cache maybe stored locally in a former version where password was not required yet?

**Comment by eddiemuc** [comment]
> I stored a quick-offline-log first and now try to compose and send. So could it be the case "edit log"?

I tried that and got the same result (password box visible).

"CREATE" in the above context means that you're in the process of creating an offline log which has not yet been sent to the server. State should still be "CREATE" even when you saved a draft before and re-enter this draft. EDIT would refer to someone editing a log which was previously send and stored on server side.

**Comment by Lineflyer** [comment]
Tried to reproduce it on another device in the same manner I did on the problematic device, but was also now unable to reproduce.
On the problematic device I refreshed the cache several times due to this problem but the problem persisted. Now I refreshed again and suddenly the password field is shown.

I have no idea how to reproduce.
Lets close until I encounter the problem again.

Thanks @eddiemuc any sorry for the noise.
